### ISSUE TITLE
Docker image crashing repeatedly on Raspberry Pi

### ISSUE THREAD

**Issue by samjewell** [MiguelNdeCarvalho/speedtest-exporter#78]
Running on Raspberry Pi B

And raspbian
VERSION="10 (buster)"

Output looks like this:
```
pi@raspberrypi:~ $ time curl localhost:9798
curl: (7) Failed to connect to localhost port 9798: Connection refused

real	0m1.589s
user	0m0.142s
sys	0m0.111s
pi@raspberrypi:~ $ sudo docker ps
CONTAINER ID   IMAGE                                  COMMAND                  CREATED              STATUS                            PORTS     NAMES
88c3e0943d3b   miguelndecarvalho/speedtest-exporter   "python -u exporter.…"   About a minute ago   Restarting (139) 15 seconds ago             speedtest-exporter
pi@raspberrypi:~ $ time curl localhost:9798
curl: (56) Recv failure: Connection reset by peer

real	0m3.523s
user	0m0.099s
sys	0m0.090s
pi@raspberrypi:~ $ sudo docker ps
CONTAINER ID   IMAGE                                  COMMAND                  CREATED         STATUS                            PORTS                                       NAMES
88c3e0943d3b   miguelndecarvalho/speedtest-exporter   "python -u exporter.…"   2 minutes ago   Up 5 seconds (health: starting)   0.0.0.0:9798->9798/tcp, :::9798->9798/tcp   speedtest-exporter
pi@raspberrypi:~ $ time curl localhost:9798
curl: (7) Failed to connect to localhost port 9798: Connection refused

real	0m0.202s
user	0m0.107s
sys	0m0.078s
pi@raspberrypi:~ $ sudo docker ps
CONTAINER ID   IMAGE                                  COMMAND                  CREATED         STATUS                            PORTS     NAMES
88c3e0943d3b   miguelndecarvalho/speedtest-exporter   "python -u exporter.…"   2 minutes ago   Restarting (139) 22 seconds ago             speedtest-exporter
```

**Comment by MiguelNdeCarvalho** [comment]
Hey,

Can you try to get the logs? I think it should be related to #62 

Thanks,
MiguelNdeCarvalho

**Comment by MiguelNdeCarvalho** [comment]
Hey,

Can you try to get the logs? I think it should be related to #62

Thanks,
MiguelNdeCarvalho

**Comment by samjewell** [comment]
# TLDR

- No logs
- Warning that the image's platform is armv7, but my platform is armv6 (RPi v1)
- I have a workaround, which is not using docker at all, and running python directly.

# Long version

I am seeing a warning as follows:
```
WARNING: The requested image's platform (linux/arm/v7) does not match the detected host platform (linux/arm/v6) and no specific platform was requested
```

There are no logs:
```
pi@raspberrypi:~ $ docker logs speedtest-exporter
pi@raspberrypi:~ $
```

Docker events looks like this, on a continuous loop:
```
pi@raspberrypi:~ $ docker events
2021-06-30T09:51:23.778925430+01:00 network disconnect b8026c28780fcf75e19737cc7600c3c7d2862e0752ed9b3cd0726e0e0abf476a (container=0e791bd55696cbd5b969fcb58d46b4e02811a95fcef5f350f9baeaa6a2d01540, name=host, type=host)
2021-06-30T09:51:26.589892912+01:00 network connect b8026c28780fcf75e19737cc7600c3c7d2862e0752ed9b3cd0726e0e0abf476a (container=0e791bd55696cbd5b969fcb58d46b4e02811a95fcef5f350f9baeaa6a2d01540, name=host, type=host)
2021-06-30T09:51:30.322737244+01:00 container start 0e791bd55696cbd5b969fcb58d46b4e02811a95fcef5f350f9baeaa6a2d01540 (image=miguelndecarvalho/speedtest-exporter, name=speedtest-exporter)
2021-06-30T09:51:30.743273017+01:00 container die 0e791bd55696cbd5b969fcb58d46b4e02811a95fcef5f350f9baeaa6a2d01540 (exitCode=139, image=miguelndecarvalho/speedtest-exporter, name=speedtest-exporter)
2021-06-30T09:51:31.264414191+01:00 network disconnect b8026c28780fcf75e19737cc7600c3c7d2862e0752ed9b3cd0726e0e0abf476a (container=0e791bd55696cbd5b969fcb58d46b4e02811a95fcef5f350f9baeaa6a2d01540, name=host, type=host)
```

I tried adding `--network host` when running the container, but that didn't help (as someone in #62 suggested that fix).


I can't even start a shell in the container - I've tried the following, as per [this blog post](https://medium.com/@pimterry/5-ways-to-debug-an-exploding-docker-container-4f729e2c0aa8), but it still crashes immediately
```
pi@raspberrypi:~ $ docker run -it --entrypoint="" speedtest-exporter sh
WARNING: The requested image's platform (linux/arm/v7) does not match the detected host platform (linux/arm/v6) and no specific platform was requested
pi@raspberrypi:~ $
```

But I was able to workaround the issue by running directly with Python outside Docker (ie not use Docker at all) on my Raspberry Pi v1 - in case anyone else is trying this too.

**Comment by MiguelNdeCarvalho** [comment]
Hey,

Right now the docker image only supports `amd64`, `arm64` and `arm/v7` architectures. Should we add `arm/v6` for older architectures?

Thanks,
MiguelNdeCarvalho

**Comment by samjewell** [comment]
> Should we add arm/v6 for older architectures?

Yeah that sounds good to me 👍

**Comment by samjewell** [comment]
![image](https://user-images.githubusercontent.com/2903904/125843054-09ccfde6-aeae-4d5a-a763-de320f7854d0.jpeg)

I just bought a new Raspberry Pi- version 4. But I still see the docker container crashing and restarting over and over again :-(

**Comment by MiguelNdeCarvalho** [comment]
Hey,

Are you running the 32 bits? Can you give me the output of `uname -m` please?

Thanks,
MiguelNdeCarvalho

**Comment by htd-tstepien** [comment]
Same issue here. But with a twist.

uname -m
```
armv7l
```
docker compose (I try to run this [docker compose](https://github.com/dipan29/internet-monitoring.git) but the only container with exporter is crashing - due to `restart always` it's restarting over and over)
```
service:
   speedtest:
    tty: true
    stdin_open: true
    expose:
      - 9798
    ports:
      - 9798:9798
    image: miguelndecarvalho/speedtest-exporter
    restart: always
    networks:
      - back-tier
   ```
   
   And I get instant restarts too

**Comment by MiguelNdeCarvalho** [comment]
Hey,

@htd-tstepien as I referenced here #71 the problem is related to using RaspberryOS 32 bits.

Thanks,
MiguelNdeCarvalho

**Comment by htd-tstepien** [comment]
Oh shut! Thanks, that's huge, so it's not connected to that, sorry then for misunderstanding my issue!

**Comment by MiguelNdeCarvalho** [comment]
> Oh shut! Thanks, that's huge, so it's not connected to that, sorry then for misunderstanding my issue! 

No problem, unfortunately they haven't pushed the fix yet

**Comment by rababb2** [comment]
@MiguelNdeCarvalho Any update on getting support for 32-bit?

**Comment by MiguelNdeCarvalho** [comment]
Hey,

@rababb2 I hope that I can do that still today. Can't you migrate to arm64?

Thanks,
MiguelNdeCarvalho

**Comment by rababb2** [comment]
@MiguelNdeCarvalho I don't think I have a 64-bit raspberry pi at this time.... It's a 3B+.

**Comment by MiguelNdeCarvalho** [comment]
@rababb2 you can get the Ubuntu Server 20.04 LTS 64 bit from here https://ubuntu.com/download/raspberry-pi

**Comment by MiguelNdeCarvalho** [comment]
Hey,

This problem should have been fixed on the latest release v3.3.4 . I will close the issue, but feel free to reopen if you find this behavior again.

Thanks,
MiguelNdeCarvalho
